arwen/contexts/metering.go

```diff
@@ -78,38 +76,52 @@ func (context *meteringContext) BoundGasLimit(value int64) uint64 {
 	return limit
 }
 
-// DeductAndLockGasIfAsyncStep will deduct the gas for an async step and also lock gas for the callback, if the execution is an asynchronous call
-func (context *meteringContext) DeductAndLockGasIfAsyncStep() error {
-	context.gasLockedForAsyncStep = 0
-
-	input := context.host.Runtime().GetVMInput()
-	if input.CallType != vmcommon.AsynchronousCall {
-		return nil
-	}
-
+// UseGasForAsyncStep consumes the AsyncCallStep gas cost on the currently
+// running Wasmer instance
+// TODO the warmInstance will be affected if this function is misused
```

Comment:
warm instance is run only in scQuery mode - thus we should not concern about it. warm instance gets the gasliimit reset at every call.

Reply:
Yes, forgot to delete this comment. Added as a note to self, then committed it accidentally.